models/oci/oci.go

```diff
@@ -112,3 +120,106 @@ func getLayerMediaType(extension string) types.MediaType {
 	}
 	return types.MediaType(fmt.Sprintf("%s.%s", oci.CanonicalMediaTypePrefix, extension))
 }
+
+// function to pull models from any OCI-compatible repository
+func PushToOCIRegistry(dirPath, registryAdd, repositoryAdd, imageTag, username, password string) error {
+
+	// Create a new file store
+	fs, err := file.New(".")
+	if err != nil {
+		return ErrWriteFile(err)
+	}
+
+	ctx := context.Background()
+
+	mediaType := "application/vnd.test.folder"
+	fileNames := []string{dirPath}
+	fileDescriptors := make([]v1.Descriptor, 0, len(fileNames))
+	for _, name := range fileNames {
+		fileDescriptor, err := fs.Add(ctx, name, mediaType, "")
+		if err != nil {
+			return ErrFileNotFound(err)
+		}
```

Comment:
ErrAddLayer?
Please ensure the errors are according to the actions being performed.

Reply:
Done.